xio-core/src/main/java/com/xjeffrose/xio/firewall/ConnectionLimiter.java

```diff
@@ -1,37 +1,49 @@
-package com.xjeffrose.xio.server;
+package com.xjeffrose.xio.firewall;
 
+import static com.codahale.metrics.MetricRegistry.name;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.MetricRegistry;
+import com.xjeffrose.xio.server.XioServerLimits;
 import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelFutureListener;
 import io.netty.channel.ChannelHandler;
 import io.netty.channel.ChannelHandlerContext;
 import java.util.concurrent.atomic.AtomicInteger;
 import lombok.extern.slf4j.Slf4j;
 
-@ChannelHandler.Sharable
 @Slf4j
-public class XioConnectionLimiter extends ChannelDuplexHandler {
-
+@ChannelHandler.Sharable
+public class ConnectionLimiter extends ChannelDuplexHandler {
   private final AtomicInteger numConnections;
   private final int maxConnections;
+  private final Counter connections;
 
-  public XioConnectionLimiter(int maxConnections) {
-    this.maxConnections = maxConnections;
+  public ConnectionLimiter(MetricRegistry metrics, XioServerLimits limits) {
+    this.maxConnections = Math.min(limits.maxConnections(), 15000);
```

Comment:
should we clamp this to a range rather than just limiting to a 15000 max?

Reply:
limits.maxConnections is a config value - we're just limiting how high you can configure it

Reply:
btw - there used to be two channel handlers (one for the config value and one for the internal limit)
IMO - limiting here is better